packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -519,3 +447,118 @@ export abstract class SharedObject<TEvent extends ISharedObjectEvents = ISharedO
 
     protected abstract applyStashedOp(content: any): unknown;
 }
+
+/**
+ * SharedObject with simplified, synchronous summarization and GC
+ */
+export abstract class SharedObject<TEvent extends ISharedObjectEvents = ISharedObjectEvents>
+    extends SharedObjectCore<TEvent> {
+    /**
+     * True while we are summarizing this object's data.
+     */
+    private _isSummarizing: boolean = false;
+
+    protected get serializer(): IFluidSerializer {
+        /**
+         * During summarize, the SummarySerializer keeps track of IFluidHandles that are serialized. These handles
+         * represent references to other Fluid objects and are used for garbage collection.
+         *
+         * This is fine for now. However, if we implement delay loading in DDss, they may load and de-serialize content
+         * in summarize. When that happens, they may incorrectly hit this assert and we will have to change this.
+         */
+        assert(!this._isSummarizing,
+            0x075 /* "SummarySerializer should be used for serializing data during summary." */);
+        return super.serializer;
+    }
+
+    /**
+     * @param id - The id of the shared object
+     * @param runtime - The IFluidDataStoreRuntime which contains the shared object
+     * @param attributes - Attributes of the shared object
+    */
+    constructor(
```

Comment:
I believe you can remove constructor and TSC will be happy to find one on SharedObjectCore